relayer/pkg/chainlink/txm/nonce.go

```diff
@@ -64,6 +64,23 @@ func (nm *nonceManager) HealthReport() map[string]error {
 	return map[string]error{nm.Name(): nm.starter.Healthy()}
 }
 
+func (nm *nonceManager) Sync(ctx context.Context, address *felt.Felt, chainId string, client NonceManagerClient) error {
+	if err := nm.validate(address, chainId); err != nil {
```

Comment:
oh nice, i didn't realize validate was also locking. i'll leave for now but good catch

Reply:
actually, i think it is better to leave it as is because confirmer and broadcaster both access the nonce manager concurrently. Therefore, even if you check that the lock is held in a hypothetical "validateLocked", you don't know whether the lock was held by the caller of "validateLocked" or by another goroutine. 

If it's by another goroutine, then we could be reading the lock when the other goroutine is changing it.

Reply:
i could just always call validate() within the bounds of the caller's lock. maybe that's what you were referring to all along